connectors/sources/sharepoint.py

```diff
@@ -238,14 +241,105 @@ async def get_content(
             convert_to_b64,
             source=source_file_name,
         )
-        async with aiofiles.open(file=source_file_name, mode="r") as target_file:
+        return {
+            "_id": document.get("id"),
+            "_timestamp": document.get("_timestamp"),
+            "_attachment": await self.convert_file_to_b64(source_file_name),
+        }
+
+    async def get_site_page_for_online(self, site_url, filename):
+        """Get metadata of site pages for SharePoint Online
+
+        Args:
+            site_url (str): Site path of sharepoint
+            filename (str): Name of file
+
+        Returns:
+            data: API response
+        """
+        response = await anext(
+            self._api_call(
+                url_name=LIST_BY_TITLE,
+                host_url=self.host_url,
+                parent_site_url=site_url,
+                filename=filename,
+            )
+        )
+        return response["value"][0].get("CanvasContent1")  # pyright: ignore
+
+    async def convert_file_to_b64(self, source_file_name):
+        """This method convert th file content into b64
+        Args:
+            source_file_name: Name of source file
+        Returns:
+            attachment_content: Attachment content in b64
+        """
+        async with aiofiles.open(file=source_file_name) as target_file:
             # base64 on macOS will add a EOL, so we strip() here
             attachment_content = (await target_file.read()).strip()
-        await remove(source_file_name)  # pyright: ignore
+        try:
+            await remove(source_file_name)  # pyright: ignore
+        except Exception as exception:
+            logger.warning(
+                f"Could not remove file from: {source_file_name}. Error: {exception}"
+            )
```

Comment:
Nit: I don't think `from` is necessary here, unless we are talking about a directory.

```suggestion
            logger.warning(
                f"Could not remove file: {source_file_name}. Error: {exception}"
            )
```